### DISCUSSION TITLE
Scope of built-in vars (eg .Code, .Root)

### DISCUSSION THREAD

**Discussion by simsrw73** [JanDeDobbeleer/oh-my-posh#1529] [Ideas]
Are these vars scoped only to their respective segments? It would be hugely convenient to use some vars is other segments in order to conserver screen space. Eg. To have a small shell segment that changes to red background based on exit code. Or an OS segment that is set to purple background when .Root. Is this possible?

Also, if features change from version to version of Oh My Posh is there a way to declare in the theme which version is required for the theme to work?

**Comment by JanDeDobbeleer** [comment]
They're only scoped inside the title and some others as adding these is de facto rather expensive. At least with the current design. Ley me see if we can revisit that.

**Reply by simsrw73** [reply]
Thanks for your response and for looking into this. I'm loving my new prompt and hate that it took me so long to learn about OMP.

**Reply by JanDeDobbeleer** [reply]
@simsrw73 check #1565

**Reply by simsrw73** [reply]
Cool! Really like this-it's a lot more flexible and allows for more economy of space by mixing multiple segments into one. 

A few places in the docs give lists of variables: Title, Text segment. Wonder if these could be consolidated in the Config-Segment doc?

I think there is some overlap in variable names available in some segments now: eg User & UserName in the Session segment. Don't know if that can be fixed without breaking stuff though.

Showing my PS ignorance, I did not figure out how to test while using the PowerShell install of OMP on Windows. I had to disable it and switch to installing with scoop and overwriting with the version I built from source. Was there some simple way to test with the PowerShell install method? Overwriting the executable didn't work: it was automatically replaced on launching. OMP puts itself at the head of $env:PATH, and I couldn't figure out how to get my test build to be used.

One last thing I just tested and didn't get to work--probably, because I'm rushing through this while I'm supposed to be working my day-job: using .Env.VarName in a text segment always errors. Am I doing something wrong here: 

    {
        "type": "text",
        "style": "plain",
        "foreground": "transparent",
        "background": "#cccccc",
        "properties": {
            "text": "{{ .Env.OS }}"
        }
    }

**Reply by JanDeDobbeleer** [reply]
No that should in theory work if that env var is available.
